source/Patches/EndGameSummary.cs

```diff
@@ -0,0 +1,141 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using UnityEngine;
+using TownOfUs.Roles;
+using TownOfUs.Roles.Modifiers;
+using HarmonyLib;
+
+namespace TownOfUs.Patches
+{
+    [HarmonyPatch]
+    static class EndGameSummary
+    {
+        static class AdditionalTempData
+        {
+            public static List<PlayerInfo> playerData = new List<PlayerInfo>();
+
+            //leaving this in as i assume it maybe needed in future, feel free to remove if you think otherwise.
+            public static void clear()
+            {
+                playerData.Clear();
+            }
+
+            internal class PlayerInfo
+            {
+                public byte PlayerId { get; set; }
+                public string PlayerName { get; set; }
+                public Faction Faction { get; set; }
+                public Role CurrentRole { get; set; }
+                public Modifier Modifier { get; set; }
+                public int TasksCompleted { get; set; }
+                public int TasksTotal { get; set; }
+            }
+        }
+
+        public static void UpdatePlayerInfo()
+        {
+            Il2CppSystem.Collections.Generic.List<GameData.PlayerInfo> players = GameData.Instance.AllPlayers;
+
+            var pinfolist = new List<AdditionalTempData.PlayerInfo>();
```

Comment:
Nit: I prefer using fully qualified types over `var`; I find them much easier to read and reason about.

Reply:
i agree, makes it easier to read, but var makes changing code easier, as all the variables change with the assignment.

i have changed this to the fully qualified type as i dont expect this code to change anymore.

Reply:
I don't think "speed of rewriting the code" is the important thing to optimize. Your code has to be read by a human many more times than it is modified by a human, so we should optimize for the thing which happens more often.